src/rp2040/hardware_structs/include/hardware/structs/dma.h

```diff
@@ -8,46 +10,173 @@
+    // Alias for channel 0 TRANS_COUNT register
     io_rw_32 al1_transfer_count_trig;
+
+    _REG_(DMA_CH0_AL2_CTRL_OFFSET)
+    // Alias for channel 0 CTRL register
     io_rw_32 al2_ctrl;
+
+    _REG_(DMA_CH0_AL2_TRANS_COUNT_OFFSET)
+    // Alias for channel 0 TRANS_COUNT register
     io_rw_32 al2_transfer_count;
+
+    _REG_(DMA_CH0_AL2_READ_ADDR_OFFSET)
+    // Alias for channel 0 READ_ADDR register
     io_rw_32 al2_read_addr;
+
+    _REG_(DMA_CH0_AL2_WRITE_ADDR_TRIG_OFFSET)
+    // Alias for channel 0 WRITE_ADDR register
     io_rw_32 al2_write_addr_trig;
+
+    _REG_(DMA_CH0_AL3_CTRL_OFFSET)
+    // Alias for channel 0 CTRL register
     io_rw_32 al3_ctrl;
+
+    _REG_(DMA_CH0_AL3_WRITE_ADDR_OFFSET)
+    // Alias for channel 0 WRITE_ADDR register
     io_rw_32 al3_write_addr;
+
+    _REG_(DMA_CH0_AL3_TRANS_COUNT_OFFSET)
+    // Alias for channel 0 TRANS_COUNT register
     io_rw_32 al3_transfer_count;
+
+    _REG_(DMA_CH0_AL3_READ_ADDR_TRIG_OFFSET)
+    // Alias for channel 0 READ_ADDR register
     io_rw_32 al3_read_addr_trig;
-}  dma_channel_hw_t;
+} dma_channel_hw_t;
 
 typedef struct {
-    dma_channel_hw_t ch[NUM_DMA_CHANNELS];
-    uint32_t _pad0[16 * (16 - NUM_DMA_CHANNELS)];
+    dma_channel_hw_t ch[12];
+
+    uint32_t _pad0[64];
+
+    _REG_(DMA_INTR_OFFSET)
+    // Interrupt Status (raw)
+    // 0x0000ffff [0-15]  : INTR (0): Raw interrupt status for DMA Channels 0
     io_ro_32 intr;
+
+    _REG_(DMA_INTE0_OFFSET)
+    // Interrupt Enables for IRQ 0
+    // 0x0000ffff [0-15]  : INTE0 (0): Set bit n to pass interrupts from channel n to DMA IRQ 0
     io_rw_32 inte0;
+
+    _REG_(DMA_INTF0_OFFSET)
+    // Force Interrupts
+    // 0x0000ffff [0-15]  : INTF0 (0): Write 1s to force the corresponding bits in INTE0
     io_rw_32 intf0;
+
+    _REG_(DMA_INTS0_OFFSET)
+    // Interrupt Status for IRQ 0
+    // 0x0000ffff [0-15]  : INTS0 (0): Indicates active channel interrupt requests which are currently causing IRQ 0 to be asserted
     io_rw_32 ints0;
-    uint32_t _pad1[1];
+
+    uint32_t _pad1;
+
+    _REG_(DMA_INTE1_OFFSET)
+    // Interrupt Enables for IRQ 1
+    // 0x0000ffff [0-15]  : INTE1 (0): Set bit n to pass interrupts from channel n to DMA IRQ 1
     io_rw_32 inte1;
+
+    _REG_(DMA_INTF1_OFFSET)
+    // Force Interrupts for IRQ 1
+    // 0x0000ffff [0-15]  : INTF1 (0): Write 1s to force the corresponding bits in INTE0
     io_rw_32 intf1;
+
+    _REG_(DMA_INTS1_OFFSET)
+    // Interrupt Status (masked) for IRQ 1
+    // 0x0000ffff [0-15]  : INTS1 (0): Indicates active channel interrupt requests which are currently causing IRQ 1 to be asserted
     io_rw_32 ints1;
+
+    _REG_(DMA_TIMER0_OFFSET)
     io_rw_32 timer[4];
-    io_wo_32 multi_channel_trigger;
+
+    _REG_(DMA_MULTI_CHAN_TRIGGER_OFFSET)
+    // Trigger one or more channels simultaneously
+    // 0x0000ffff [0-15]  : MULTI_CHAN_TRIGGER (0): Each bit in this register corresponds to a DMA channel
+    io_rw_32 multi_channel_trigger;
```

Comment:
The datasheet PDF says that this is SC (self-clearing) so arguably `io_wo_32` was more correct than `io_rw_32`?
And I checked and the SVD has `<modifiedWriteValues>clear</modifiedWriteValues>` (see https://www.keil.com/pack/doc/CMSIS/SVD/html/elem_registers.html#elem_register) for `MULTI_CHAN_TRIGGER`, so I guess theoretically this _could_ still be `io_wo_32`? Entirely up to you @kilograham if you feel that the distinction and effort are worth it though!

Reply:
it sounds like it is read/write but always reads as zero, no? in which case I guess `io_rw_32` is fine... if the SVD set it was wrtie only then we'd use wo